### DISCUSSION TITLE
No support for Gemini API?

### DISCUSSION THREAD

**Discussion by dillfrescott** [open-webui/open-webui#4742] [General]

**Comment by justinh-rahb** [comment]
Supported with either Functions in WebUI:
https://openwebui.com/f/justinrahb/google_genai/

Or in the [Pipelines server](https://github.com/open-webui/pipelines):
https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/google_manifold_pipeline.py

**Reply by kali-777** [reply]
> Supported with either Functions in WebUI: https://openwebui.com/f/justinrahb/google_genai/

I'm sorry I'm also more like Sysadmin then programmer I used to do Pascal/C++ long ago, now very basic Python 😎
SO as per Gemini-2.0-Flash looks like Ur function 'sadly' using OLD Google GenAI API NOT a NEW/"FREE" version Google AI Studio Gemini API! 🤔
"Limitations of Open WebUI Function: The Open WebUI function you linked (https://openwebui.com/f/justinrahb/google_genai) specifically uses the Google GenAI API, not the Gemini API. These are two different services. GenAI is an older API, while Gemini is a newer experimental service with more advanced capabilities. 
**Important Note:** The GenAI API is deprecated and may eventually become unavailable."

[Lite-LLM-Gemini](https://docs.litellm.ai/docs/providers/gemini) they support out-of-the-box, but looks like not fully...
[[Can't use gemini vision when using external litellm proxy](https://github.com/open-webui/open-webui/discussions/3806#top)

Pretty please would be possible to create other/NEW version of Open-WebUI Function for "FREE" Gemini API! 🤗 

Many thanks

**Comment by dillfrescott** [comment]
Oh, thank you

**Comment by kali-777** [comment]
Looks like Google itself "Solved" the problem [Google Gemini Open AI PROXY](https://github.com/google-gemini/proxy-to-gemini) 🤓

**Reply by bhanubais** [reply]
But how to use it with Open webUI?

**Reply by kali-777** [reply]
This will create an Open AI API Endpoint which You can use it inside Open webUI then it will PROXY the Requests for Gemini, https://github.com/google-gemini/proxy-to-gemini/blob/main/README.md

Copilot 😁😜


### Steps to Run Proxy-to-Gemini in Docker
1. **Obtain a Gemini API Key**:
   - Get your API key from the AI Studio.

2. **Run the Docker Container**:
   ```bash
   docker run -p 5555:5555 -e GEMINI_API_KEY=<YOUR_API_KEY> google-gemini/proxy-to-gemini
   ```

### Integrate with OpenWebUI
1. **Pull the OpenWebUI Docker Image**:
   ```bash
   docker pull ghcr.io/open-webui/open-webui:main
   ```

2. **Run OpenWebUI with Docker**:
   ```bash
   docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
   ```

3. **Configure OpenWebUI to Use Proxy-to-Gemini**:
   - Open the OpenWebUI configuration file.
   - Set the API endpoint to point to your proxy server:
     ```yaml
     api:
       url: http://localhost:5555/v1/chat/completions
     ```

4. **Restart OpenWebUI**:
   ```bash
   docker restart open-webui
   ```

**Comment by MillionthOdin16** [comment]
Considering it's offered for free, you can also just use openrouter by adding it as an openai compatible endpoint

**Reply by kali-777** [reply]
yeah, U can, but personally in Lite LLM i have more "faith"/trust than OpenRouter, i really do NOT want to share & **filter** ALL my Chats with even more Companies 😁

**Comment by Tixx7** [comment]
Gemini works by just using their openai api compatible endpoint: 

```
https://generativelanguage.googleapis.com/v1beta/openai/
```

**Reply by venttarc** [reply]
> Gemini works by just using their openai api compatible endpoint:
> 
> ```
> https://generativelanguage.googleapis.com/v1beta/openai/
> ```

Yo sweetie, thank you so much.

**Comment by cellulosa** [comment]
is there a way to setup Gemini via env variables? Or does it require manually adding `https://generativelanguage.googleapis.com/v1beta/openai/` and the api key under admin/settings/connections/Manage OpenAI API Connections?
